createurl.php

```diff
@@ -31,6 +31,8 @@ function bve($bv) {
     }
 }
 
+$_POST['urlr'] = $_POST['naked'] . "/?utm_source=" . $_POST['utm-source'] . "&utm_medium=" . $_POST['utm-medium'] . "&utm_campaign=" . $_POST['utm-name'];
```

Comment:
Doing it like this is a bad idea.
What happens if the user doesn't specify the a campaign? The link gets "corrupt" because it still appends the campaign parameters.

You should be doing a check if all (or one) of the campaign parameters was actually entered:

```PHP
if ( ( isset( $_POST['utm-source'] ) && !empty( $_POST['utm-source'] ) ) ||
     ( isset( $_POST['utm-medium'] ) && !empty( $_POST['utm-medium'] ) ) ||
     ( isset( $_POST['utm-source'] ) && !empty( $_POST['utm-source'] ) ) ) {
        $_POST['urlr'] = $_POST['naked'] . "/?utm_source=" . $_POST['utm-source'] . "&utm_medium=" . $_POST['utm-medium'] . "&utm_campaign=" . $_POST['utm-name'];
}
```

Or something like this, it's probably possible to do it shorter then this. For example you could also append the parameters one by one, though I never used campaign parameters so I don't know if all of them are always necessary.